drivers/i2c/i2c_emul.c

```diff
@@ -74,13 +83,96 @@ static int i2c_emul_get_config(const struct device *dev, uint32_t *dev_config)
 	return 0;
 }
 
+#ifdef CONFIG_I2C_TARGET
+static int i2c_emul_send_to_target(const struct device *dev, struct i2c_msg *msgs, uint8_t num_msgs)
+{
+	struct i2c_emul_data *data = dev->data;
+	const struct i2c_target_callbacks *callbacks = data->target_cfg->callbacks;
+
+	for (uint8_t i = 0; i < num_msgs; ++i) {
+		LOG_DBG("    msgs[%u].flags? 0x%02x", i, msgs[i].flags);
+		if ((msgs[i].flags & I2C_MSG_RW_MASK) == I2C_MSG_READ) {
+			for (uint32_t j = 0; j < msgs[i].len; ++j) {
+				int rc = 0;
+
+				if (j == 0) {
+					LOG_DBG("    Calling read_requested with data %p",
+						(void *)&msgs[i].buf[j]);
+					rc = callbacks->read_requested(data->target_cfg,
+								       &msgs[i].buf[j]);
+				} else {
+					LOG_DBG("    Calling read_processed with data %p",
+						(void *)&msgs[i].buf[j]);
+					rc = callbacks->read_processed(data->target_cfg,
+								       &msgs[i].buf[j]);
+				}
```

Comment:
As a follow up can you add support for `CONFIG_I2C_TARGET_BUFFER_MODE`?

Reply:
Sure, added it to my TODO list